config/v1/types_cluster_version.go

```diff
@@ -215,6 +222,71 @@ type UpdateHistory struct {
 // ClusterID is string RFC4122 uuid.
 type ClusterID string
 
+// ClusterVersionCapability enumerates optional, core cluster components.
+type ClusterVersionCapability string
+
+const (
+	// ClusterVersionCapabilityOpenShiftSamples manages the sample
+	// image streams and templates stored in the openshift
+	// namespace, and any registry credentials, stored as a secret,
+	// needed for the image streams to import the images they
+	// reference.
+	ClusterVersionCapabilityOpenShiftSamples ClusterVersionCapability = "openshift-samples"
+)
+
+// ClusterVersionCapabilitySet defines sets of cluster version capabilities.
+type ClusterVersionCapabilitySet string
+
+const (
+	// ClusterVersionCapabilitySetNone is an empty set containing no capabilities.
+	ClusterVersionCapabilitySetNone ClusterVersionCapabilitySet = "None"
+
+	// ClusterVersionCapabilitySet4_11 is the recommended set of
+	// optional capabilities for the 4.11 version of OpenShift.
+	// This list will remain the same no matter which version of
+	// OpenShift is installed.
+	ClusterVersionCapabilitySet4_11 ClusterVersionCapabilitySet = "v4.11"
+)
+
+// ClusterVersionCapabilitySets defines sets of cluster version capabilities.
+var ClusterVersionCapabilitySets = map[ClusterVersionCapabilitySet]ClusterVersionCapability{
+	ClusterVersionCapabilitySetNone: {},
+	ClusterVersionCapabilitySet4_11: {
+		ClusterVersionCapabilityOpenShiftSamples,
+	},
+}
+
+// ClusterVersionCapabilitiesSpec selects the managed set of
+// optional, core cluster components.
+// +k8s:deepcopy-gen=true
+type ClusterVersionCapabilitiesSpec struct {
+	// baselineCapabilitySet selects an initial set of
+	// capabilities, which can be extended via
+	// additionalEnabledCapabilities.
+	BaselineCapabilitySet ClusterVersionCapabilitySet `json:"baselineCapabilitySet,omitempty"`
+
+	// additionalEnabledCapabilities add additional capabilies to
+	// the managed set.
+	// +listType=atomic
+	// +optional
+	AdditionalEnabledCapabilities []ClusterVersionCapability `json:"additionalEnabledCapabilities,omitempty"`
+}
+
+// ClusterVersionCapabilitiesStatus describes the state of optional,
+// core cluster components.
+// +k8s:deepcopy-gen=true
+type ClusterVersionCapabilitiesStatus struct {
+	// include lists all the capabilities that are currently included.
+	// +listType=atomic
+	// +optional
+	Include []string `json:"include,omitempty"`
+
+	// exclude lists all the capabilities that are currently excluded.
+	// +listType=atomic
+	// +optional
+	Exclude []string `json:"exclude,omitempty"`
```

Comment:
please remove

Reply:
Removed, now that 224e90ed pivots to [your suggested][1] `enabledCapabilities` and `knownCapabilities`.

[1]: https://github.com/openshift/api/pull/1106#discussion_r799819632